beacon-chain/core/altair/block.go

```diff
@@ -0,0 +1,125 @@
+	if err != nil {
+		return nil, err
+	}
+
+	currentSyncCommittee, err := state.CurrentSyncCommittee()
+	if err != nil {
+		return nil, err
+	}
+	committeeKeys := currentSyncCommittee.Pubkeys
+	votedKeys := make([]bls.PublicKey, 0, len(committeeKeys))
+	votedIndices := make([]types.ValidatorIndex, 0, len(committeeKeys))
+
+	// Verify sync committee signature.
+	for i := uint64(0); i < sync.SyncCommitteeBits.Len(); i++ {
+		if sync.SyncCommitteeBits.BitAt(i) {
+			pubKey, err := bls.PublicKeyFromBytes(committeeKeys[i])
+			if err != nil {
+				return nil, err
+			}
+			votedKeys = append(votedKeys, pubKey)
+			votedIndices = append(votedIndices, committeeIndices[i])
+		}
+	}
+	ps := helpers.PrevSlot(state.Slot())
+	d, err := helpers.Domain(state.Fork(), helpers.SlotToEpoch(ps), params.BeaconConfig().DomainSyncCommittee, state.GenesisValidatorRoot())
+	if err != nil {
+		return nil, err
+	}
+	pbr, err := helpers.BlockRootAtSlot(state, ps)
+	if err != nil {
+		return nil, err
+	}
+	sszBytes := p2pType.SSZBytes(pbr)
+	r, err := helpers.ComputeSigningRoot(&sszBytes, d)
+	if err != nil {
+		return nil, err
+	}
+	sig, err := bls.SignatureFromBytes(sync.SyncCommitteeSignature)
+	if err != nil {
+		return nil, err
+	}
+	if !sig.FastAggregateVerify(votedKeys, r) {
+		return nil, errors.New("could not verify sync committee signature")
+	}
+
+	// Calculate sync committee and proposer rewards
+	activeBalance, err := helpers.TotalActiveBalance(state)
+	if err != nil {
+		return nil, err
+	}
+	totalActiveIncrements := activeBalance / params.BeaconConfig().EffectiveBalanceIncrement
+	totalBaseRewards := baseRewardPerIncrement(activeBalance) * totalActiveIncrements
+	maxParticipantRewards := totalBaseRewards * params.BeaconConfig().SyncRewardWeight / params.BeaconConfig().WeightDenominator / uint64(params.BeaconConfig().SlotsPerEpoch)
+	participantReward := maxParticipantRewards / params.BeaconConfig().SyncCommitteeSize
+	proposerReward := participantReward * params.BeaconConfig().ProposerWeight / (params.BeaconConfig().WeightDenominator - params.BeaconConfig().ProposerWeight)
+
+	// Apply sync committee rewards.
+	earnedProposerReward := uint64(0)
+	for _, index := range votedIndices {
+		if err := helpers.IncreaseBalance(state, index, participantReward); err != nil {
+			return nil, err
+		}
+		earnedProposerReward += proposerReward
+	}
+	// Apply proposer rewards.
+	proposerIndex, err := helpers.BeaconProposerIndex(state)
+	if err != nil {
+		return nil, err
+	}
+	if err := helpers.IncreaseBalance(state, proposerIndex, earnedProposerReward); err != nil {
+		return nil, err
+	}
+
+	return state, nil
+}
+
+// This returns the base reward per increment for the beacon state.
+//
+// def get_base_reward_per_increment(state: BeaconState) -> Gwei:
+//    return Gwei(EFFECTIVE_BALANCE_INCREMENT * BASE_REWARD_FACTOR // integer_squareroot(get_total_active_balance(state))
+func baseRewardPerIncrement(activeBalance uint64) uint64 {
```

Comment:
Putting this hear for now but we might move this to other places once we implement other things that use this